internal/cmdrpkgupdate/discover.go

```diff
@@ -50,31 +50,48 @@ func (r *runner) discoverUpdates(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	var updates [][]string
+	var upstreamUpdates [][]string
+	downstreamUpdatesMap := make(map[string][]porchapi.PackageRevision) // map from the upstream package revision to a list of its downstream package revisions
+
```

Comment:
Would it be easier to just create separate functions for the upstream and downstream cases here? The switch statement inside the loop will always have the same value for every iteration. I think it would also remove the need for two separate switch statements (finding updates and printing the results).

Reply:
done! Thanks for the suggestion, the code is a lot cleaner this way.